### DISCUSSION TITLE
Validation resets active(routeIs()) navigation button

### DISCUSSION THREAD

**Discussion by adamvaclav** [livewire/livewire#8236] [1. Help] [answered]
Hi all!

I am having a little trouble with navigation button that loses its active state when validation happens in my livewire component.
(I tried googling my mistake and even tried some AI help but no solution in sight, I'm laravel/livewire newbie so maybe you'll spot the mistake right away).

Basically, every time validation happens, if its live validation or validation inside updateTeam method, this validation button loses its active state. Route is not changing at all so I have no idea what's going on.

Before:
<img width="718" alt="image" src="https://github.com/livewire/livewire/assets/35834443/b99d31ec-f847-4755-a703-1001c271a7e2">
After:
<img width="711" alt="image" src="https://github.com/livewire/livewire/assets/35834443/586c502b-ede7-4ce8-b64c-3cbf859aafc3">

This is my volt component(without html for clarity):

```
use Livewire\Volt\Component;
use Livewire\Attributes\Validate;
use Illuminate\Validation\Rule;

new class extends Component {
    
    public $team;
    #[Validate]
    public $teamname;
    #[Validate]
    public $token;
    public $old_token;


    public function mount($team) {
        $this->team = $team;
        $this->teamname = $team->teamname;
        $this->token = $team->token;
    }

    public function rules()
    {
        return [
            'teamname' => 'required|min:3|max:25|alpha_dash',
            'token' => 'required|min:3|max:25|alpha_dash|lowercase|unique:App\Models\Team,token,'.$this->team->id,
        ];
    }


    public function updateTeam() {
        $this->validate();
        $this->old_token = $this->team->token;
        $this->team->update([
            'teamname' => $this->teamname,
            'token' => $this->token
        ]);
        $this->dispatch('teamupdated');
        if ($this->token != $this->old_token) {
            session()->flash('success', 'Page was refreshed to match the new token url');
            return $this->redirect('/team/' . $this->token . '/settings');
        }
    }
```

And my component html:

```
<div class="space-y-1 last:pb-6">
    <x-poker.team-header :team="$team" />
    <x-poker.section-card header="Team settings">
        // Team settings form
    </x-poker.section-card>
</div>
```

And this is navigation link inside component poker.team-header:

```<x-poker.form.link-icon-button url="/team/{{ $team->token }}/settings" :active="request()->routeIs('settings')">```

**Comment by joshhanley** [comment, marked as answer]
@adamvaclav the problem is that whenever you make a Livewire request, the request is made to the `/livewire/update` route. So what happens then is that the `request()->routeIs('settings')` check will return `false` as the route is no longer the settings route.

There are a couple of ways to fix that. The simplest is, if your button doesn't need to do anything or be changed by Livewire, you could put a `wire:ignore` on it.

If that is no good though, you could store the original route name in a public property and check against that instead of  using the `routeIs` method.

Hope this helps!

**Reply by adamvaclav** [reply]
Great answer, I didn't know about request being made to livewire/update. Thank you very much.

**Reply by joshhanley** [reply]
@adamvaclav No worries! 🙂 hmm yeah we should probably have something in the docs that explains the request process 🤔

**Reply by danie-ramdhani** [reply]
if you really need active the link using laravel:

```php
(request()->is('livewire/update') ? str_replace(url('/'), '', url()->previous()) : str_replace(url('/'), '', request()->url())) === '/foo'
```

i think using alpine is the best solution (alternative):

```blade
<a href="/foo" :class="{ 'active': window.location.pathname === $el.getAttribute('href') }">Link</a>
```
